packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -175,6 +258,45 @@ export default class ScriptTransformer {
     return undefined;
   }
 
+  private async _getTransformerAsync(filename: Config.Path) {
+    if (!this._config.transform || !this._config.transform.length) {
+      return null;
+    }
+
+    const transformPath = this._getTransformPath(filename);
+
+    if (!transformPath) {
+      return null;
+    }
+
+    const cached = this._transformCache.get(transformPath);
+    if (cached) {
+      return cached;
+    }
+
+    let transformer: AsyncTransformer = await import(transformPath);
```

Comment:
we should have a mutex here. Just a `Map<Config.Path, Promise<void>>` where the `Path` is `transformPath` and we can `await` that above the `this._transformCache.get` call. If we get a cache miss, we add a promise to the map there. that way the cache will be populated once.

checking if the map has a promise before `await`ing will avoid the extra tick as well.

(make sure the promise we stick in the cache resolve to something empty so we don't keep a reference to something we don't need)

Reply:
Modified, the Promise only gets resolved after cache has been set. A bit curious whether we can make cache and the mutex single source of truth while allowing transformer sync loading to work as-is 🤔